### DISCUSSION TITLE
Nuxt2 Get Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/vercel__launcher.launcher' imported from /opt/rust/nodejs.js

### DISCUSSION THREAD

**Discussion by charlie0228** [vercel/community#6547] [Help] [answered]
### Summary

I using Nuxt2 with config below, and got `ERR_MODULE_NOT_FOUND` error
However, the same code I've deployed 3 days ago, it works fine.

Here's my url for work fine version: https://tw-stock-news-ok2vjlbv5-azureblues-projects.vercel.app/
And the failed version: https://tw-stock-news-36exgtmdn-azureblues-projects.vercel.app/

### Steps to Reproduce

`vercel.json`
```
{
  "version": 2,
  "builds": [
    {
      "src": "nuxt.config.ts",
      "use": "@nuxtjs/vercel-builder"
    }
  ]
}
```
`Error Message`
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/vercel__launcher.launcher' imported from /opt/rust/nodejs.js
at new NodeError (node:internal/errors:387:5)
at finalizeResolution (node:internal/modules/esm/resolve:330:11)
at moduleResolve (node:internal/modules/esm/resolve:907:10)
at defaultResolve (node:internal/modules/esm/resolve:1115:11)
at nextResolve (node:internal/modules/esm/loader:163:28)
at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
at ESMLoader.import (node:internal/modules/esm/loader:525:22)
at importModuleDynamically (node:internal/modules/cjs/loader:1136:29)
at importModuleDynamicallyWrapper (node:internal/vm/module:438:21) {
code: 'ERR_MODULE_NOT_FOUND'
}
Node.js process exited with exit status: 1. The logs above can help with debugging the issue.
INIT_REPORT Init Duration: 120.84 ms	Phase: init	Status: error	Error Type: Runtime.ExitError
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/vercel__launcher.launcher' imported from /opt/rust/nodejs.js
at new NodeError (node:internal/errors:387:5)
at finalizeResolution (node:internal/modules/esm/resolve:330:11)
at moduleResolve (node:internal/modules/esm/resolve:907:10)
at defaultResolve (node:internal/modules/esm/resolve:1115:11)
at nextResolve (node:internal/modules/esm/loader:163:28)
at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
at ESMLoader.import (node:internal/modules/esm/loader:525:22)
at importModuleDynamically (node:internal/modules/cjs/loader:1136:29)
at importModuleDynamicallyWrapper (node:internal/vm/module:438:21) {
code: 'ERR_MODULE_NOT_FOUND'
}
Node.js process exited with exit status: 1. The logs above can help with debugging the issue.
INIT_REPORT Init Duration: 157.05 ms	Phase: invoke	Status: error	Error Type: Runtime.ExitError
Unknown application error occurred
```

**Comment by Eddie344** [comment]
I have the same problem

**Reply by swarnava** [reply]
Can you share deployment url?

**Reply by Eddie344** [reply]
@swarnava https://procomfort-rljct0ela-eduard-bazhenovs-projects.vercel.app/

**Comment by ardenateam** [comment]
I have just get that error and need help with it, cause it affected our production site

**Reply by swarnava** [reply]
Can you share deployment url?

**Reply by UmarovJX** [reply]
@swarnava 
https://www.ardena.uz/ - is latest
https://adrena-site-keco33l94-ardenateamversel-gmailcom.vercel.app/ru - latest working deployment

**Reply by UmarovJX** [reply]
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/vercel__launcher.launcher' imported from /opt/rust/nodejs.js
at new NodeError (node:internal/errors:387:5)
at finalizeResolution (node:internal/modules/esm/resolve:330:11)
at moduleResolve (node:internal/modules/esm/resolve:907:10)
at defaultResolve (node:internal/modules/esm/resolve:1115:11)
at nextResolve (node:internal/modules/esm/loader:163:28)
at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
at ESMLoader.import (node:internal/modules/esm/loader:525:22)
at importModuleDynamically (node:internal/modules/cjs/loader:1136:29)
at importModuleDynamicallyWrapper (node:internal/vm/module:438:21) {
code: 'ERR_MODULE_NOT_FOUND'
}
Node.js process exited with exit status: 1. The logs above can help with debugging the issue.
INIT_REPORT Init Duration: 166.21 ms	Phase: init	Status: error	Error Type: Runtime.ExitError
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/var/task/vercel__launcher.launcher' imported from /opt/rust/nodejs.js
at new NodeError (node:internal/errors:387:5)
at finalizeResolution (node:internal/modules/esm/resolve:330:11)
at moduleResolve (node:internal/modules/esm/resolve:907:10)
at defaultResolve (node:internal/modules/esm/resolve:1115:11)
at nextResolve (node:internal/modules/esm/loader:163:28)
at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
at ESMLoader.import (node:internal/modules/esm/loader:525:22)
at importModuleDynamically (node:internal/modules/cjs/loader:1136:29)

@swarnava full error log

**Reply by swarnava** [reply]
Thanks! It looks like your last deployment was 198 days so there were lot of changes happened in the meantime. If you can share a [minimal reproducible ](https://stackoverflow.com/help/minimal-reproducible-example)steps, we can take a closer look. But I don't think it's directly related to any recent changes we have made.

**Reply by UmarovJX** [reply]
i can share full project if you wish. all today changes are minor vue file updates because project is finished

**Reply by UmarovJX** [reply]
you can just check files changed in last commits to see that there where no changes that might affect deployment

**Reply by swarnava** [reply]
Hi, we don't have access to private repository unfortunately .

**Reply by UmarovJX** [reply]
@swarnava hey i just created new app and deployed it using Vercel CLI
create app:
npm init nuxt-app temp-app

with unversal rendering mode and Deployment target: Server (Node.js hosting)

add vercel.json
```
{
    "builds": [
        {
            "src": "nuxt.config.js",
            "use": "@nuxtjs/vercel-builder",
            "config": {}
        }
    ]
}
```

deploy it using vercel command and i got the same problem

**Reply by swarnava** [reply]
You can get start from here: https://vercel.com/clone?b=main&id=67753070&s=https%3A%2F%2Fgithub.com%2Fvercel%2Fvercel%2Ftree%2Fmain%2Fexamples%2Fnuxtjs&showOptionalTeamCreation=false&template=nuxtjs

**Reply by UmarovJX** [reply]
@swarnava I am getting 404 at your link

**Reply by swarnava** [reply]
Can you try this https://vercel.com/new/clone?b=main&id=67753070&s=https%3A%2F%2Fgithub.com%2Fvercel%2Fvercel%2Ftree%2Fmain%2Fexamples%2Fnuxtjs&showOptionalTeamCreation=false&template=nuxtjs ?

**Reply by UmarovJX** [reply]
@swarnava 
your link is nuxt 3 version, we have problems with nuxt 2

**Reply by swarnava** [reply]
Thanks. Unfortunately at this point we need a [minimal repository](https://stackoverflow.com/help/minimal-reproducible-example) to reproduce this. I cannot find any recent changes which may cause this issue

**Reply by Jakhongiri2max** [reply]
@swarnava does not my message https://github.com/orgs/vercel/discussions/6547#discussioncomment-9154842 give a reproduction of problem? just created an brand new nuxt2 app and it fails to worl

**Reply by Jakhongiri2max** [reply]
I am sorry for multiple accounts, i am also getting confused between teams

**Reply by UmarovJX** [reply]
@swarnava i think you can use https://github.com/UmarovJX/error-reproduction as minimal repo. just empty nuxt2 app.
deployed at https://error-reproduction.vercel.app/ and fails with the error

**Comment by swarnava** [comment]
@charlie0228  can you still reproduce with redeploy?

**Reply by charlie0228** [reply]
Yes, after redeploy, the previous worked commit became failed.

**Reply by swarnava** [reply]
I can see the last deployment was 6 hrs ago. Can you redeploy now and let us know?

**Reply by charlie0228** [reply]
I redeploy again, and it still got failed.
https://tw-stock-news-cw3oo4rmr-azureblues-projects.vercel.app/

**Reply by swarnava** [reply]
Can you redeploy an working commit such as this https://vercel.com/azureblues-projects/tw-stock-news/75rzPyhhH94tpsX5TTWT9YUABKG9?

**Reply by charlie0228** [reply]
Sorry, I would like to redeploy this version but I got this message.
<img width="526" alt="image" src="https://github.com/vercel/community/assets/24387287/ee119d6b-9cda-4a5e-aff5-4a2334446942">

**Reply by charlie0228** [reply]
Also, I cannot click `Promote` and `Instant Rollback` because I am a Hobby customer....

**Reply by charlie0228** [reply]
Is this issue related to the node.js version?
My project is using node.js v16

**Reply by swarnava** [reply]
I believe the latest update of `@nuxt/vercel-builder` is causing some regression. Can you pin it to previous version in `vercel.json` and see if it helps?

```js
{
  "version": 2,
  "builds": [
    {
      "src": "nuxt.config.js",
      "use": "@nuxtjs/vercel-builder@0.24.0"
    }
  ]
}
```

**Reply by CrownsMedia2013** [reply]
Same issue here. I have changed the node.js version several times, but nothing works so far.

**Reply by swarnava** [reply]
Are you able to reproduce with `@nuxtjs/vercel-builder@0.24.0` as well?

**Reply by charlie0228** [reply]
@swarnava I've already tried it when I meet this issue, let me try it again.

**Reply by charlie0228** [reply]
Still not work after I lock the version `@nuxtjs/vercel-builder@0.24.0`.
https://tw-stock-news-fjyl5do92-azureblues-projects.vercel.app/

**Reply by swarnava** [reply]
Thanks. Unfortunately at this point we need a [minimal repository](https://stackoverflow.com/help/minimal-reproducible-example) to reproduce this. I cannot find any recent changes which may cause this issue

**Comment by swarnava** [comment]
Here's how I just deployed Nuxt 2 package in Vercel: 

- Created skeleton using  `npx create-nuxt-app nuxt2-app`
-  Updated vercel.json using this: https://v2.nuxt.com/deployments/vercel/#ssr-with-vercel
- added `"node-fetch-native": "^1.4.1"` in `dependencies` of package.json. Not sure why it's required but I noticed it is suggested by Community: https://github.com/nuxt/vercel-builder/issues/805
- set  `"engines": {"node": "18.x"}` in package.json

At this point, I will recommend check with Nuxt community to find out more information about the specific error.

**Reply by UmarovJX** [reply]
I have repeated your steps
repo https://github.com/UmarovJX/error2
deployment  https://error2.vercel.app/
still getting the same error

**Reply by Eddie344** [reply]
Still error: https://procomfort-git-dev-eduard-bazhenovs-projects.vercel.app/

**Reply by charlie0228** [reply]
Still got error with your solution.

**Comment by LiamThursfield** [comment]
Not got anything too helpful to add here unfortunately. Just that I have the same problem.

So here is a very basic Nuxt2 site deployed a couple weeks ago (working): https://cordelectrics-three.vercel.app/
Here is the exact same commit, but on a different project, deployed today (server error as above): https://cordelectrics-r5k8.vercel.app/:

**Comment by juanmarinelli** [comment]
I have the same problem with 2 sites. I made some changes and uploaded them and it broke. https://tienda-nutridiet.vercel.app/
I have other sites with the same setup uploaded a few weeks ago and they work fine.
I tried with "use": "@nuxtjs/vercel-builder@0.24.0" but it also gives an error

**Comment by QuiiBz** [comment, marked as answer]
I reverted a change that we made recently and seems to impact the Nuxt builder. We will investigate to fix the issue and make sure it doesn’t reproduce in the future.
You can redeploy your applications and it should work again as expected.

**Reply by juanmarinelli** [reply]
perfect! all works now! thanks

**Reply by Eddie344** [reply]
it's working! thanks!

**Reply by LiamThursfield** [reply]
All good here too - thanks for that!

**Reply by QuiiBz** [reply]
Quick update: we re-applied the reverted change along with a fix, and tested the Nuxt 2 builder (`@nuxtjs/vercel-builder`). Everything should still work as expected.
